src/jit/compiler.cpp

```diff
@@ -2478,6 +2479,28 @@ void Compiler::compSetProcessor()
 #endif // _TARGET_AMD64_
             }
         }
+#else  // !FEATURE_CORECLR
+        // If this is not FEATURE_CORECLR, the flags for SSE3, SSSE3, SSE41 and SSE42 are collectively
+        // implied by EnableSSE3_4, and, since we don't support the HW intrinsics, these and the AVX
+        // ISAs are the only ones supported by the VM and relied upon by the JIT.
+        //
+        if (JitConfig.EnableSSE3_4())
+        {
+            opts.setSupportedISA(InstructionSet_SSE3);
+            opts.setSupportedISA(InstructionSet_SSE41);
+            opts.setSupportedISA(InstructionSet_SSE42);
+
+            if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_AVX) && JitConfig.EnableAVX())
```

Comment:
Should this just be a check for `AVX2`?

`S.N.Vector` doesn't differentiate between just AVX or AVX and AVX2 support and it would better match what we were doing before the HWIntrinsic support was added: https://github.com/dotnet/coreclr/blob/release/2.0.0/src/jit/compiler.cpp#L2468

Reply:
I think the main differentiation here would be whether `canUseVexEncoding` returns `true` even when `Vector<T>` is 16-bytes, which might be beneficial in some cases; but I am unsure if it meets the desktop "back-compat" bar.

Reply:
Good point. I'll just check for AVX2 (and add SSSE3)